command/sub8_alarm/sub8_alarm/alarm_helpers.py

```diff
@@ -0,0 +1,80 @@
+import rospy
+from sub8_msgs.msg import Alarm
+from std_msgs.msg import Header
+import json
+
+class AlarmBroadcaster(object):
+    def __init__(self):
+        '''Alarm Broadcaster'''
+        # Notify the world when our alarm happens
+        self.alarm_pub = rospy.Publisher('/alarm', Alarm, latch=True, queue_size=10)
+
+        # Notify the world that our alarm has spawned
+        # For now, unused - future: Use for system reports
+        # self.alarm_spawn_pub = rospy.Publisher('/alarm_spawn', Alarm, latch=True, queue_size=1)
+
+        # Get ROS name
+        self.node_name = rospy.get_name()
+
+        # Alarms associated with this broadcaster
+        # In the future, there will be some background bookkeeping
+        self.alarms = []
+
+    def add_alarm(self, name, action_required=False, severity=5, problem_description=None, parameters=None):
+        '''Factory method for creating a new alarm'''
+        new_alarm = AlarmRaiser(
+            alarm_name=name,
+            node_name=self.node_name,
+            alarm_publisher=self.alarm_pub,
+            action_required=action_required,
+            problem_description=problem_description,
+            parameters=parameters
+        )
+
+        self.alarms.append(new_alarm)
+        return new_alarm
+
+
+class AlarmRaiser(object):
+    def __init__(self, alarm_name, node_name, alarm_publisher, action_required=False, severity=5, problem_description=None, parameters=None):
+        '''Alarm object, does alarms
+        Generates alarm messages
+        '''
+        assert severity in range(0, 5 + 1), "Severity must be an integer between 0 and 5"
+        self._alarm_name = alarm_name
+        self._node_name = node_name
+        self._action_required = action_required
+        self._severity = severity
+        self._problem_description = problem_description
+        self._parameters = parameters
+
+        self._alarm_pub = alarm_publisher
+
+    def raise_alarm(self, problem_description=None, parameters=None):
+        '''Arguments are override parameters'''
+        assert (problem_description != "") or (self._problem_description is not None), "No problem description has been set for this alarm"
```

Comment:
Do we want to leave this assert here later on? Good for debugging, but during production If an alarm doesn't have a problem description, isn't it better just to let it through (since the alarm will still go off and be handled)?